concepts/05 UI Components/DataGrid/15 Columns/40 Customize Cells/3 Customize Row Apperance.md

```diff
@@ -0,0 +1,147 @@
+                        // To override alternation color
+                        e.rowElement.removeClass("dx-row-alt");
+                    }
+                }
+            }
+        });
+    });
+
+##### Angular
+
+    <!-- tab: app.component.html -->
+    <dx-data-grid
+        (onRowPrepared)="onRowPrepared($event)">
+    </dx-data-grid>
+
+    <!-- tab: app.component.ts -->
+    // ...
+    export class AppComponent {
+        onRowPrepared(e) {
+            if (e.rowType === "data") {
+                if (e.data.Velocity > e.data.SpeedLimit) {
+                    e.cellElement.style.cssText = "color: white; background-color: red";
+                    // or
+                    e.rowElement.classList.add("my-class");
+                    // To override alternation color
+                    e.rowElement.className = e.rowElement.className.replace("dx-row-alt", "");
+                }
+            }
+        }
+    }
+
+##### Vue
+
+    <!-- tab: App.vue -->
+    <template>
+        <DxDataGrid ...
+            @row-prepared="onRowPrepared">
+        </DxDataGrid>
+    </template>
+
+    <script>
+    import 'devextreme/dist/css/dx.light.css';
+
+    import DxDataGrid from 'devextreme-vue/data-grid';
+
+    export default {
+        components: {
+            DxDataGrid
+        },
+        methods: {
+            onRowPrepared(e) {
+                if (e.rowType === "data") {
+                    if (e.data.Velocity > e.data.SpeedLimit) {
+                        e.cellElement.style.cssText = "color: white; background-color: red";
+                        // or
+                        e.rowElement.classList.add("my-class");
+                        // To override alternation color
+                        e.rowElement.className = e.rowElement.className.replace("dx-row-alt", "");
+                    }
+                }
+            }
+        }
+    }
+    </script>
+
+##### React
+
+    <!-- tab: App.js -->
+    import React from 'react';
+
+    import 'devextreme/dist/css/dx.light.css';
+
+    import DataGrid from 'devextreme-react/data-grid';
+
+    const rowPrepared = (e) => {
+        if (e.rowType === "data") {
+            if (e.data.Velocity > e.data.SpeedLimit) {
+                e.cellElement.style.cssText = "color: white; background-color: red";
+                // or
+                e.rowElement.classList.add("my-class");
+                // To override alternation color
+                e.rowElement.className = e.rowElement.className.replace("dx-row-alt", "");
+            }
+        }
+    }
+
+    function App() {
+        return (
+            <DataGrid ...
+                onRowPrepared={rowPrepared}>
+            </DataGrid>
+        );
+    }
+
+    export default App;
+    
+---
+
+You can also use the [dataRowTemplate](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/#dataRowTemplate) property to customize row appearance.
+
+To change row selection color, use the following CSS rules:
```

Comment:
I'm not sure if these CSS rules have something to do with the **dataRowTemplate** property or did you move on to something completely separate. Maybe we can break this topic down into some sections? Or just add some text that makes it clear that this is the same or different approach.